### PR TITLE
 Updates how .swift-version file is validated to prevent it from being needed in Objective-C pods

### PR DESCRIPTION
**Pull request by atreat**
Updates the check for the .swift-version file to only perform this check if the targets include swift sources. 

Addresses #6742

### CODE REVIEW

**Comment by atreat** [comment]
It looks like my text editor stripped a bunch of trailing whitespace from the CHANGELOG.md file.

If necessary can try to revert that change so the diff is a little cleaner

**Comment by dnkoutso** [comment]
Yes please the `CHANGELOG` intentionally has the two spaces after each change. Please revert entirely the file and add your entry at the top.

File: lib/cocoapods/validator.rb [hunk 1]
```diff
@@ -82,6 +82,8 @@ def validate
       perform_linting
       perform_extensive_analysis(a_spec) if a_spec && !quick
 
+
```
Comment (dnkoutso) [inline lib/cocoapods/validator.rb]: extra lines, revert this change.

File: spec/unit/validator_spec.rb [hunk 2]
```diff
@@ -870,46 +870,56 @@ def test_swiftpod
...
+        result = validator.results.first
+        result.type.should == :warning
+        result.message.should == "The validator for " \
           'Swift projects uses Swift 3.0 by default, if you are using a ' \
           'different version of swift you can use a `.swift-version` file ' \
           'to set the version for your Pod. For example to use Swift 2.3, ' \
           "run: \n    `echo \"2.3\" > .swift-version`"
       end
 
+
```
Comment (dnkoutso) [inline spec/unit/validator_spec.rb]: extra lines, revert

File: lib/cocoapods/validator.rb [hunk 3]
```diff
@@ -382,6 +378,18 @@ def validate_documentation_url(spec)
       validate_url(spec.documentation_url) if spec.documentation_url
     end
 
+    def validate_dot_swift_version
+      if used_swift_version != nil && dot_swift_version.nil?
```
Comment (dnkoutso) [inline lib/cocoapods/validator.rb]: can this be `if !used_swift_version.nil?`?

File: spec/unit/validator_spec.rb [hunk 4]
```diff
@@ -870,46 +870,56 @@ def test_swiftpod
...
         validator = test_swiftpod
+        validator.stubs(:dot_swift_version).returns('3.1.0')
+        validator
+      end
+
+      it 'fails on deployment target < iOS 8 for Swift Pods' do
+        validator = test_swiftpod_with_dot_swift_version
         validator.validate
 
+        validator.results.inspect
```
Comment (dnkoutso) [inline spec/unit/validator_spec.rb]: needed?

File: spec/unit/validator_spec.rb [hunk 5]
```diff
@@ -870,46 +870,56 @@ def test_swiftpod
         validator
       end
 
-      it 'fails on deployment target < iOS 8 for Swift Pods' do
+      def test_swiftpod_with_dot_swift_version
```
Comment (dnkoutso) [inline spec/unit/validator_spec.rb]: I would change that to:

```ruby
test_swiftpod_with_dot_swift_version(version = '3.1.0')
...
```

**Comment by dnkoutso** [comment]
I think you need a spec that validates that the absence of swift version does not display the message.

**Comment by dnkoutso** [comment]
Also run `rubocop -a` in your command line to fix potential code style issues etc automatically.

**Comment by atreat** [comment]
👍  Thanks for the feedback (it all makes sense). I'll push up the fixes to this asap

File: CHANGELOG.md [hunk 6]
```diff
@@ -16,6 +16,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
   [Dimitris Koutsogiorgas](https://github.com/dnkoutso)
   [#1698](https://github.com/CocoaPods/CocoaPods/issues/1698) 
 
+* Prevents need for .swift-version file in Objective-C pods  
+ +  [Austin Emmons](https://github.com/atreat)
```
Comment (dnkoutso) [inline CHANGELOG.md]: nit extra space here

Comment (atreat) [inline CHANGELOG.md]: Whats the reasoning behind this changelog trailing whitespace? I'm curious to know why that is the desired format. (Added in c44c8ff)

Comment (dnkoutso) [inline CHANGELOG.md]: I think it has to do with formatting. I was commenting mostly that the second line was not aligned with the top line.

Comment (dnkoutso) [inline CHANGELOG.md]: Still need to fix this before approval

Comment (atreat) [inline CHANGELOG.md]: I get it now, looking at the rendered file makes more sense. I didn't release a double space at the end of a line in markdown is a way to force a newline.

**Comment by dnkoutso** [comment]
@atreat nice! can you please squash to a single commit? then its done!

**Comment by atreat** [comment]
Awesome, thanks for the review. Squashed it.
